File: recursions/fibonacci/fibonacci.php
<?php
function fibanacciIterative($n) {
    $fib = array(0, 1);
    
    for ($i = 2; $i <= $n; ++$i) {
        array_push($fib, $fib[$i-1] + $fib[$i-2]);
    }

    return $fib[$n];
}

$iterative = fibanacciIterative(6);
echo "Iterative: " . $iterative . "\n";

function fibanacciRecursion($n) {
    if ($n < 2) {
        return $n;
    }

    return fibanacciRecursion($n-1) + fibanacciRecursion($n-2);
}

$recursion = fibanacciRecursion(7);
echo "Iterative: " . $recursion . "\n";

File: recursions/fibonacci/fibonacci.lua
local function fibonacciIterative(n)
    n = n + 1
    local fib = {0, 1}

    for i = 3, n do
        table.insert(fib, fib[i-1] + fib[i-2])
    end

    return fib[n]
end

local iterative = fibonacciIterative(8)
print("Iterative: " .. iterative)

local function fibonacciRecursion(n)
    if n < 2 then
        return n
    end

    return fibonacciRecursion(n-1) + fibonacciRecursion(n-2)
end

local recursion = fibonacciRecursion(9)
print("Recursive: " .. recursion)
